cmd/registry/registry.go

```diff
@@ -0,0 +1,368 @@
+			DarknodeRegistryAddress: ethereum.DarknodeRegistryAddressOnRopsten.String(),
+		}
+	case "kovan":
+		config = ethereum.Config{
+			Network:                 ethereum.NetworkKovan,
+			URI:                     "https://kovan.infura.io",
+			RepublicTokenAddress:    ethereum.RepublicTokenAddressOnKovan.String(),
+			DarknodeRegistryAddress: ethereum.DarknodeRegistryAddressOnKovan.String(),
+		}
+	default:
+		log.Fatal("unrecognized network name")
+	}
+
+	auth := bind.NewKeyedTransactor(key.PrivateKey)
+	auth.GasPrice = big.NewInt(5000000000)
+
+	client, err := ethereum.Connect(config)
+	if err != nil {
+		log.Fatal("fail to connect to ethereum")
+	}
+
+	return dnr.NewDarknodeRegistry(context.Background(), client, auth, &bind.CallOpts{})
+}
+
+func RegisterAll(registry dnr.DarknodeRegistry) error {
+
+	conf, err := loadConfig("./deployment.json")
+	if err != nil {
+		return errors.New("could not read file deployment.json")
+	}
+
+	for i := range conf.Configs {
+		address := conf.Configs[i].Config.Address
+		pk, err := crypto.BytesFromRsaPublicKey(&conf.Configs[i].Config.Keystore.RsaKey.PublicKey)
+		if err != nil {
+			log.Fatal("cannot get rsa public key ", err)
+		}
+
+		// Check if node has already been registered
+		isRegistered, err := registry.IsRegistered(address)
+		if err != nil {
+			return fmt.Errorf("[%v] %sCouldn't check node's registration%s: %v\n", address, red, reset, err)
+		}
+
+		// Register the node if not registered
+		if !isRegistered {
+			minimumBond, err := registry.MinimumBond()
+			if err != nil {
+				return err
+			}
+
+			_, err = registry.Register(address.ID(), pk, &minimumBond)
+			if err != nil {
+				return fmt.Errorf("[%v] %sCouldn't register node%s: %v\n", address, red, reset, err)
+			} else {
+				log.Printf("[%v] %sNode will be registered next epoch%s\n", address, green, reset)
+			}
+		} else if isRegistered {
+			log.Printf("[%v] %sNode already registered%s\n", address, yellow, reset)
+		}
+	}
+
+	return nil
+
+}
+
+// DeregisterAll takes a slice of republic private keys and registers them
+func DeregisterAll(addresses []string, registry dnr.DarknodeRegistry) error {
+	conf, err := loadConfig("./deployment.json")
+	if err != nil {
+		return errors.New("could not read file deployment.json")
+	}
+
+	for i := range conf.Configs {
+		address := conf.Configs[i].Config.Address
+
+		// Check if node has already been registered
+		isRegistered, err := registry.IsRegistered(address)
+		if err != nil {
+			return fmt.Errorf("[%v] %sCouldn't check node's registration%s: %v\n", address, red, reset, err)
+		}
+
+		if isRegistered {
+			_, err = registry.Deregister(address.ID())
+			if err != nil {
+				return fmt.Errorf("[%v] %sCouldn't deregister node%s: %v\n", address, red, reset, err)
+			} else {
+				log.Printf("[%v] %sNode will be deregistered next epoch%s\n", address, green, reset)
+			}
+		} else if isRegistered {
+			log.Printf("[%v] %sNode already registered%s\n", address, yellow, reset)
```

Comment:
`else if isRegistered` should either be `!isRegistered` or just an `else`, and the log should say already **deregistered**